packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -42,30 +43,45 @@ export interface DrugOrderFormProps {
   onCancel: () => void;
 }
 
-const comboSchema = z.object({
+const comboSchema = {
   value: z.string(),
   valueCoded: z.string(),
   default: z.boolean().optional(),
-});
-
-const medicationOrderFormSchema = z.object({
-  isFreeTextDosage: z.boolean(),
-  freeTextDosage: z.string(),
-  dosage: z.number(),
-  unit: comboSchema,
-  route: comboSchema,
+};
+
+const schemaFields = {
+  freeTextDosage: z.string().refine((value) => value !== '', { message: 'Add free dosage note' }),
+  dosage: z.number({ invalid_type_error: 'A dosage is required' }),
+  unit: z.object({ ...comboSchema }, { invalid_type_error: 'Please select a unit' }),
+  route: z.object({ ...comboSchema }, { invalid_type_error: 'Please select a route' }),
```

Comment:
I want to get this PR in but it would be great if we could translate these strings. Maybe in a follow-up?

Reply:
yeah sure  a follow up will be best. Am still thinking of a way to translate this since I can't use the `useTranslation` hook right here

Reply:
You can't just move these definitions into the component?

Reply:
You can use https://github.com/openmrs/openmrs-esm-core/blob/d13f7278fc4ea65b1267284b9560231d7b596987/packages/framework/esm-utils/src/translate.ts#L4

What do you @brandones ?

```
translateFrom(
"@openmrs/esm-patient-medications-app",
  'selectARoute',
  "Please select a route"
)
```

Do remember to add `t("selectARoute", "Please select a route")` in comments on the top of the line for the i18 parser to add the key in the translation file.

Reply:
Yeah, would probably be easier just to move the definition into the component, and then you wouldn't have to mess with the comments. But that should work too.